### PR TITLE
Feature/cu kb87y4 keyboard&game controller UI handling

### PR DESCRIPTION
**Pull request by devildrake**
This PR establishes Navigation functionalities by keyboard and Game Controller, pressing Enter/A on a selected button/Toggle will press it, doing so on a Slider will "Lock control" on it, changing it's color, allowing horizontal input to modify the slider value and blocking navigation until Enter/A is pressed again to confirm the slider value,

### CODE REVIEW

**Review by GBMiro — COMMENTED** [review]
Can't test it until tomorrow afternoon / night with controllers but looks good! I'll test it and approve it if everything works fine

**Review by JaimeBea — DISMISSED** [review]

**Review by jcarlos0305 — APPROVED** [review]

**Review by GBMiro — COMMENTED** [review]
Sensitivity setting does not work when using the controller (seems it's being completely ignored and uses 100 always).

Slider using the keyboard lock/unlock with Enter works perfectly and you can change the value.
When you lock the slider and then click in another UI element with the mouse, the slider keeps looking like it's locked for a value change (Visual bug. The selected element changes correctly).

Also trying to lock the slider with controller sometimes does not work.

The toggle does not check/uncheck with the keyboard nor the controller

I tried to reproduce everything in the folowwing gif. All input is keyboard / controller but the first click on the toggle

ControllerUI

**Review by GBMiro — DISMISSED** [review]
Tested it again with keyboard and controller and everythings behaves as expected. Good job!

**Review by GBMiro — APPROVED** [review]

File: Project/Source/Components/UI/ComponentEventSystem.cpp [hunk 1]
```diff
@@ -42,34 +42,60 @@ void ComponentEventSystem::Update() {
...
+			selectionDir = float2(-1.f, 0.f);
+			keyPressed = true;
+		}
+		if (App->input->GetKey(SDL_SCANCODE_RIGHT) == KS_DOWN) {
+			selectionDir = float2(1.f, 0.f);
+			keyPressed = true;
+		}
+
+		PlayerController* controller = nullptr;
+		if (controller = App->input->GetPlayerController(0)) {
```
Comment (JaimeBea) [inline Project/Source/Components/UI/ComponentEventSystem.cpp:69]: You could move the assignment to the previous line to make it clearer.

File: Project/Source/Components/UI/ComponentSlider.cpp [hunk 2]
```diff
@@ -238,6 +270,16 @@ float ComponentSlider::GetNormalizedValue() const {
...
 }
 
+void ComponentSlider::ModifyValue(float mulitplier) {
+#if GAME
+	currentValue = Max(Min(currentValue + mulitplier * App->time->GetDeltaTime() * sliderSensitivity, maxValue), minValue);
+#else
+	currentValue = Max(Min(currentValue + mulitplier * App->time->GetRealTimeDeltaTime() * sliderSensitivity, maxValue), minValue);
+#endif
+	if (mulitplier != 0.0f)
+		OnValueChanged();
```
Comment (JaimeBea) [inline Project/Source/Components/UI/ComponentSlider.cpp:280]: Ifs like this, separated in two lines, are prone to errors. Either move OnValueChanged to the same line or add braces.

**Review by JaimeBea — DISMISSED** [review]

**Review by GBMiro — APPROVED** [review]

**Review by jcarlos0305 — APPROVED** [review]

File: Project/Source/Components/UI/ComponentEventSystem.cpp (hunk 1 above)
Comment (devildrake) [reply Project/Source/Components/UI/ComponentEventSystem.cpp:69]: True enough

File: Project/Source/Components/UI/ComponentSlider.cpp (hunk 2 above)
Comment (devildrake) [reply Project/Source/Components/UI/ComponentSlider.cpp:280]: I forgot that single line if were kinda forbidden

**Review by JaimeBea — DISMISSED** [review]

**Review by GBMiro — APPROVED** [review]

**Review by jcarlos0305 — APPROVED** [review]

**Review by FrancPS — APPROVED** [review]

**Review by JaimeBea — APPROVED** [review]

**Review by jcarlos0305 — APPROVED** [review]

**Review by FrancPS — APPROVED** [review]

**Pull request merged**